cpp/src/arrow/io/compressed_test.cc

```diff
@@ -273,6 +273,14 @@ TEST(TestSnappyOutputStream, NotImplemented) {
 }
 #endif
 
+#if !defined ARROW_WITH_ZLIB \
+  && !defined ARROW_WITH_BROTLI \
+  && !defined ARROW_WITH_LZ4 \
+  && !defined ARROW_WITH_ZSTD
+GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(CompressedInputStreamTest);
+GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(CompressedOutputStreamTest);
+#endif
```

Comment:
You'll need to define `GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST` in `arrow/testing/gtest_compat.h` for GTest versions < 1.11.